app/Http/Controllers/WalletAdjustmentController.php

```diff
@@ -15,6 +15,10 @@ public function __construct()
 
     public function balance(Wallet $wallet, Request $request)
     {
+        if (auth()->user()->id != $wallet->user_id){
+            abort(403);
+        }
+//         abort_if(auth()->user()->isNot($wallet->user_id), 403);
```

Comment:
I think we still can use policy even for these custom `balance` controller method. Or we can simply move it to the form request with the validation if we feel we will not use it again in different places.